### DISCUSSION TITLE
Any idiomatic way to set the cursor on multiline input?

### DISCUSSION THREAD

**Discussion by ckipp01** [oyvindberg/tui-scala#45] [Q&A]
I was playing around with this today and couldn't come up with an idiomatic way to achieve this. Currently we follow a pattern like this:

```scala
f.set_cursor(
  x = chunks(1).x + Grapheme(app.input).width + 1,
  y = chunks(1).y + 1
)
```
Where we based the `x` axis off the width of whatever string is inside of your `ParagraphWidget.text`. This works until you hit the end of your block, then everything sort of falls apart because the `width` method on `Grapheme` knows nothing about whether or not you're using `Wrap`, and also the `y` doesn't take that into account either since we're just doing a `+ 1` there.

Am I maybe missing an idiomatic way to do this where you have multi-line input? Or do you have any ideas what we could adjust to make this work to ensure the cursor is where it needs to be in multi-line input?

**Comment by oyvindberg** [comment]
Your guess is as good as mine here, honestly! :)

I should probably note that the implementation of `Grapheme` may differ from the rust version, not sure if a more correct version would make any difference here.

**Reply by ckipp01** [reply]
> I should probably note that the implementation of Grapheme may differ from the rust version, not sure if a more correct version would make any difference here.

Make sense. I'll dig a bit deeper here to see what I can find. I also have a couple ideas of possible just using the `ParagraphWidget` here and having a couple helper methods that could just spit out the `(x, y)` of where the cursor should be. I'll see what I can come up with. Thanks!
